src/conversion/viewconversiondispatcher.js

```diff
@@ -137,55 +136,71 @@ export default class ViewConversionDispatcher {
 	 * @fires documentFragment
 	 * @param {module:engine/view/documentfragment~DocumentFragment|module:engine/view/element~Element} viewItem
 	 * Part of the view to be converted.
-	 * @param {Object} additionalData Additional data to be passed in `data` argument when firing `ViewConversionDispatcher`
-	 * events. See also {@link ~ViewConversionDispatcher#event:element element event}.
+	 * @param {module:engine/model/schema~SchemaContextDefinition} [context=['$root']] Elements will be converted according to this context.
 	 * @returns {module:engine/model/documentfragment~DocumentFragment} Model data that is a result of the conversion process
 	 * wrapped in `DocumentFragment`. Converted marker elements will be set as that document fragment's
 	 * {@link module:engine/model/documentfragment~DocumentFragment#markers static markers map}.
 	 */
-	convert( viewItem, additionalData ) {
+	convert( viewItem, context = [ '$root' ] ) {
 		return this._model.change( writer => {
+			this.fire( 'viewCleanup', viewItem );
+
+			const consumable = ViewConsumable.createFrom( viewItem );
+
+			// Create model tree according to given context. Elements will be converted to the top element of this tree.
+			// Thanks to this schema will be able check items precisely.
+			// Top element of context tree is marked by a `isContextTree` attribute.
+			const position = contextToPosition( context, writer );
+
 			// Store writer in current conversion as a conversion API.
 			this.conversionApi.writer = writer;
 
-			this.fire( 'viewCleanup', viewItem );
+			// Create set for split elements. We need to remember this elements, because at the end of conversion
+			// we want to remove all empty elements that was created as a result of the split.
+			this.conversionApi.splitElements = new Set();
```

Comment:
I'm going to move `splitElements` from conversionApi to private `ViewConversiondispatcher` property.